core/src/com/mygdx/game/Base/SpritesPool.java

```diff
@@ -64,6 +64,13 @@ private void free(T object) {
         if (activeObjects.remove(object)) {
             freeObjects.add(object);
         }
-        System.out.println("active/free : " + activeObjects.size() + "/" + freeObjects.size());
+        //System.out.println("active/free : " + activeObjects.size() + "/" + freeObjects.size());
+    }
+
+    public void clear(){
+        activeObjects.removeAll(activeObjects);
+        freeObjects.removeAll(freeObjects);
```

Comment:
Our static analysis engine detects that your pull request includes a change of **using `removeAll` to clear a collection**. It is a bad practice. If you want to remove all elements from a collection `c`, **use `c.clear`,** not `c.removeAll(c)`. Calling `c.removeAll(c)` to clear a collection is **less clear, susceptible to errors from typos, less efficient and for some collections, might throw a ConcurrentModificationException.**